packages/create-amplify/src/package_manager_controller.ts

```diff
@@ -1,8 +1,56 @@
+import { execa as _execa } from 'execa';
+import { executeWithDebugLogger } from './execute_with_logger.js';
+import { type PackageManager } from './package_manager.js';
+
 export type DependencyType = 'dev' | 'prod';
 
-export type PackageManagerController = {
+export type PackageManagerControllerType = {
   installDependencies: (
     packageNames: string[],
     type: DependencyType
   ) => Promise<void>;
 };
+
+/**
+ *
+ */
+export class PackageManagerController implements PackageManagerControllerType {
```

Comment:
This should go back to the way it was with additional implementations of PackageManagerController for the other package managers

Reply:
I don't think that's necessary because all the package managers are pretty much the same. The only differences are already listed in packages/create-amplify/src/package_manager.ts

Reply:
Updated 40f3590bba689d2dea834f81834cc8d268add891